packages/app-desktop/gui/JoplinCloudLoginScreen.tsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useMemo, useReducer, useState } from 'react';
+import ButtonBar from './ConfigScreen/ButtonBar';
+import { _ } from '@joplin/lib/locale';
+import { clipboard } from 'electron';
+import Button, { ButtonLevel } from './Button/Button';
+const bridge = require('@electron/remote').require('./bridge').default;
+import { uuidgen } from '@joplin/lib/uuid';
+import { Dispatch } from 'redux';
+import { reducer, defaultState, generateLoginWithUniqueLoginCode, checkIfLoginWasSuccessful } from '@joplin/lib/services/joplinCloudUtils';
+import { AppState } from '../app.reducer';
+import Logger from '@joplin/utils/Logger';
+
+const logger = Logger.create('JoplinCloudLoginScreen');
+const { connect } = require('react-redux');
+
+interface Props {
+	dispatch: Dispatch;
+	joplinCloudWebsite: string;
+	joplinCloudApi: string;
+}
+
+const JoplinCloudScreenComponent = (props: Props) => {
+
+	const loginUrl = `${props.joplinCloudWebsite}/login`;
+	const applicationsUrl = `${props.joplinCloudApi}/api/applications`;
+
+	const [intervalIdentifier, setIntervalIdentifier] = useState(undefined);
+	const [state, dispatch] = useReducer(reducer, defaultState);
+
+	const uniqueLoginCode = useMemo(() => uuidgen(), []);
+
+	const periodicallyCheckForCredentials = () => {
+		if (intervalIdentifier) return;
+
+		const interval = setInterval(async () => {
+			try {
+				const response = await checkIfLoginWasSuccessful(applicationsUrl, uniqueLoginCode);
+				if (response && response.success) {
+					dispatch({ type: 'COMPLETED' });
+					clearInterval(interval);
+				}
+			} catch (error) {
+				logger.error(error);
+				dispatch({ type: 'ERROR', payload: error.message });
```

Comment:
Indeed that makes sense to use an action for the error too. We don't use `useReducer` at all but why not, it seems to make the code quite clear